app/Models/EntityMention.php

```diff
@@ -236,4 +236,63 @@ public function scopeDatagridElements(Builder $query, array $options): Builder
         }  
         return $query;
     }
+
+    /**
+     * @return string
+     * @throws Exception
+     */
+    public function getLink(): string
+    {
+        if ($this->isQuestElement()) {
+            return route('quests.quest_elements.index', [$this->questElement->quest->id, '#quest-element-' . $this->quest_element_id]);
+        } elseif ($this->isTimelineElement()) {
+            return route('timelines.show', [$this->timelineElement->timeline->id, '#timeline-element-' . $this->timeline_element_id]);
+        } elseif ($this->isPost()) {
+            return route($this->post->entity->pluralType() . '.show', [$this->post->entity->child->id, '#post-' . $this->entity_note_id]);
+        }
+        return '#';
+        
+    }
+
+    /**
+     * Get the entity link with ajax tooltip.
+     * When coming from an entity first, call this method on the entity. It avoids some back and worth.
+     * @return string
+     */
+    public function mentionLink(): string
+    {
+        if ($this->isQuestElement()) {
```

Comment:
Missing permission checks for non-admins. Page will 500 "entity is null" on most of these if permissions are set to now give visibility I believe. See how it was done in the old code where `->entity` has to be checked all the time.